pkg/controllers/user/logging/utils/embedded.go

```diff
@@ -307,70 +310,59 @@ func updateEmbeddedQuota(dep rv1beta2.DeploymentInterface, obj *v3.ClusterLoggin
+		}
+	}
+
 	selector := labels.NewSelector()
-	requirement, err := labels.NewRequirement(loggingconfig.LabelK8sApp, selection.Equals, []string{loggingconfig.EmbeddedESName})
+	requirement, err := labels.NewRequirement(loggingconfig.LabelK8sApp, selection.Equals, []string{serviceName})
 	if err != nil {
 		return "", err
 	}
 
-	espods, err := podLister.List(loggingconfig.LoggingNamespace, selector.Add(*requirement))
+	pods, err := podLister.List(loggingconfig.LoggingNamespace, selector.Add(*requirement))
 	if err != nil {
 		return "", err
 	}
-	esservice, err := serviceLister.Get(loggingconfig.LoggingNamespace, loggingconfig.EmbeddedESName)
+	if len(pods) == 0 {
+		return "", fmt.Errorf("deploying %s", serviceName)
+	}
+	pod := pods[0]
+
+	for _, cond := range pod.Status.Conditions {
+		if cond.Status == v1.ConditionFalse {
+			return "", fmt.Errorf("pod %s status %s is %s, reason: %s, message: %s", pod.Name, cond.Type, cond.Status, cond.Reason, cond.Message)
+		}
+	}
+	esservice, err := serviceLister.Get(loggingconfig.LoggingNamespace, serviceName)
 	if err != nil {
 		return "", err
 	}
 
 	if len(esservice.Spec.Ports) == 0 {
-		return "", fmt.Errorf("get service %s node port failed", loggingconfig.EmbeddedESName)
+		return "", fmt.Errorf("could not find the node port for %s", serviceName)
 	}
-	var esPort int32
+	var port int32
 	for _, v := range esservice.Spec.Ports {
 		if v.Name == "http" {
-			esPort = v.NodePort
+			port = v.NodePort
 			break
 		}
 	}
 
-	if len(espods) == 0 {
-		return "", fmt.Errorf("deploying %s", loggingconfig.EmbeddedESName)
-	}
-	espod := espods[0]
-	if espod.Status.Phase == running {
-		return fmt.Sprintf("http://%s:%v", espod.Status.HostIP, esPort), nil
-	}
-	return "", fmt.Errorf("got embedded elasticsearch pod status %s", espod.Status.Phase)
-}
-
-func getEmbeddedKibanaEndpoint(podLister rv1.PodLister, serviceLister rv1.ServiceLister) (kibanaEndpoint string, err error) {
-	selector := labels.NewSelector()
-	requirement, err := labels.NewRequirement(loggingconfig.LabelK8sApp, selection.Equals, []string{loggingconfig.EmbeddedKibanaName})
-	if err != nil {
-		return "", err
-	}
-	kibanapods, err := podLister.List(loggingconfig.LoggingNamespace, selector.Add(*requirement))
-	if err != nil {
-		return "", err
-	}
-	kibanaservice, err := serviceLister.Get(loggingconfig.LoggingNamespace, loggingconfig.EmbeddedKibanaName)
-	if err != nil {
-		return "", err
-	}
-	if len(kibanaservice.Spec.Ports) == 0 {
-		return "", fmt.Errorf("get service %s node port failed", loggingconfig.EmbeddedKibanaName)
-	}
-
-	if len(kibanapods) == 0 {
-		return "", fmt.Errorf("deploying %s", loggingconfig.EmbeddedKibanaName)
-	}
-	kibanapod := kibanapods[0]
-	if kibanapod.Status.Phase == running {
-		return fmt.Sprintf("http://%s:%v", kibanapod.Status.HostIP, kibanaservice.Spec.Ports[0].NodePort), nil
+	if pod.Status.Phase == running {
```

Comment:
why are we doing it? NodePort service exposes port on *every* node in the cluster. Traffic can hit any node:publicPort, and k8s will redirect it to any pod behind the deployment. Is there a reason why we try to limit it to only a node where specific pod runs? 

If the idea is to:
* make sure that there is at least one running pod backing up the deployment
* find at least one NodeIp where the deployment is exposed via NodePort

following can be done:

* by checking `endpoints` object for the service (it would have the same name as the service, and same namespace), and making sure that there is at least one endpoint address. K8s will automatically remove address from the endpoints if the pod goes down, or becomes non-healthy
* Pick any active node in the system when fetch GetEndpointNodeIP